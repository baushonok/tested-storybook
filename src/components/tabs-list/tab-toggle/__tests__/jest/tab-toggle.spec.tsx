import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import TabToggle from '../..';

describe('TabToggle', () => {
  const name = 'tab name';
  const index = 0;
  const clickHandler = jest.fn();

  describe('should match snapshot', () => {
    test('when active', () => {
      const tree = renderer
        .create(
          <TabToggle
            name={name}
            index={index}
            onClick={clickHandler}
            isActive
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when inactive', () => {
      const tree = renderer
        .create(<TabToggle name={name} index={index} onClick={clickHandler} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    test('should invoke click when tab is inactive', () => {
      render(<TabToggle name={name} index={index} onClick={clickHandler} />);
      const tabToggle = screen.getByTestId(`tab-toggle-${index}`);

      tabToggle.click();

      expect(clickHandler).toHaveBeenCalledTimes(1);
    });

    test('should not invoke click when tab is active', () => {
      render(
        <TabToggle name={name} index={index} onClick={clickHandler} isActive />
      );
      const tabToggle = screen.getByTestId(`tab-toggle-${index}`);

      tabToggle.click();

      expect(clickHandler).not.toHaveBeenCalled();
    });
  });
});
