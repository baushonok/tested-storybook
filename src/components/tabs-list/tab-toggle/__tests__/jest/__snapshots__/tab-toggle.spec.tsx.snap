// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabToggle should match snapshot when active 1`] = `
<button
  className="tab-toggle tab-toggle_active"
  data-testid="tab-toggle-0"
  onClick={[Function]}
  type="button"
>
  tab name
</button>
`;

exports[`TabToggle should match snapshot when inactive 1`] = `
<button
  className="tab-toggle "
  data-testid="tab-toggle-0"
  onClick={[Function]}
  type="button"
>
  tab name
</button>
`;
